### DISCUSSION TITLE
Pcloud on Raspberry4

### DISCUSSION THREAD

**Discussion by Exclution** [ptitSeb/box86#397] [Q&A]
Hello all, 

i installed box86 and tried to get the pcloud software for linux running. 
i used the link https://www.pcloud.com/de/download-free-online-cloud-file-storage.html and selected the 32bit version of pcloud.

unfortunately i get following error using box86:
![grafik](https://user-images.githubusercontent.com/86156461/122644013-2c22e400-d113-11eb-957d-48fd813c1da2.png)

by using --appimage-extract option it creates the folder: 
![grafik](https://user-images.githubusercontent.com/86156461/122644047-67251780-d113-11eb-92b0-ad5d090178d5.png)

by starting ./AppRun following message appears: 
![grafik](https://user-images.githubusercontent.com/86156461/122644064-7a37e780-d113-11eb-9ed6-3184f94a62b1.png)

so i wonder if there is a way to get pcloud started with box86...

Thanks in advance

**Comment by chills340** [comment]
have you tried to `sudo apt install libfuse2 ` ? i've also noticed there is a github with pre-built debs or you can grab the depends and compile from source. https://github.com/pcloudcom/console-client 

I haven't tried it myself because i didn't know what pcloud was prior to this post

**Reply by Exclution** [reply]
Hi, 

i tried and got following:
![grafik](https://user-images.githubusercontent.com/86156461/122645562-41037580-d11b-11eb-886c-22a67ba29bd6.png)

the solution compiling the console-client could work, but i wished to get the gui version running... :-(

**Reply by ptitSeb** [reply]
libfuse is not wrapped for now, so the native version will not be used.

**Comment by ptitSeb** [comment]
Try to use `BOX86_LOG=1 ./AppRun` to have more details about needed libs.

**Reply by Exclution** [reply]
Hi, 

this brings up Following 

Dynarec for ARM, with extension: HALF FAST_MULT EDSP NEON VFPv4 IDIVA PageSize: 4096 Box86 with Dynarec v0.2.3 681e06ce built on Jun 13 2021 00:03:18
Using default BOX86_LD_LIBRARY_PATH: /:lib/: lib32/: x86/ Using default BOX86_PATH: /bin/
Counted 45 Env var
Looking for /home/pi/Downloads/squashfs-root/pcloud Using emulated libnode.so
Using native (wrapped) libpthread.so.0 Using native (wrapped)
libgtk-3.so.0
Using native (wrapped) libgdk-3.so.0
Using native (wrapped) libgobject-2.0.so.0
Using native (wrapped) libglib-2.0.so.0
Using native (wrapped) libgio-2.0.so.0
Using native (wrapped) libgmodule-2.0.so.0
Using native (wrapped) libgdk_pixbuf-2.0.so.0
Using native (wrapped) libpangocairo-1.0.so.0
Using native (wrapped) libpango-1.0.so.0
Using native (wrapped) libatk-1.0.so.0
Using native (wrapped) libcairo.so.2
Using native (wrapped) libatk-bridge-2.0.so.0
Using native (wrapped) libSM.so.6
Using native (wrapped) libICE.so.6
Using native (wrapped) libxau.so.6 Using native (wrapped) libxcb.so.1
Using native (wrapped) libdbus-1.so.3
Using native (wrapped) libx11-xcb.so.1 Using native (wrapped) libxi.so.6
Using native (wrapped) libx11.so.6
Using native (wrapped) libXext.so.6
Using native (wrapped) libxdmcp.so.6
Using native (wrapped) libXcursor.so.1
Using native (wrapped) libXdamage.so.1
Using native (wrapped) libXrandr.so.2 Using native (wrapped) libXrender.so.1
Using native (wrapped) libXcomposite.so.1 Using native (wrapped) libxfixes.so.3
Using native (wrapped) libxtst.so.6 Using native (wrapped) libXss.so.1
Error loading needed lib libnss3.so
Error loading one of needed lib
Error: loading needed libs in elf /home/pi/Downloads/squashfs-root/pcloud

**Reply by ptitSeb** [reply]
So you need x86 version of `libnss3.so` (and a few bunch of other probably).

**Reply by Exclution** [reply]
Hi ptitSeb, thanks, 

can you once show me, how to get those? i'll install them all one by one then by myself and post, if i've succeeded, ok?

BR

**Reply by Exclution** [reply]
in any case, this didn't worked: 
![grafik](https://user-images.githubusercontent.com/86156461/122646647-84141780-d120-11eb-9e7e-4de1c87f3c4f.png)

**Reply by Exclution** [reply]
Hi all, 

now i've installed libnss3 and also the missing libnspr4. Unfortunately now im getting following by 
`BOX86_LOG=1 ./AppRun`:
![grafik](https://user-images.githubusercontent.com/86156461/122654886-986f0900-d14e-11eb-95a9-4b768a9d0ef5.png)

Full Output: [pcloud issue gtk.txt](https://github.com/ptitSeb/box86/files/6681328/pcloud.issue.gtk.txt)

Any Suggestions?
Thanks in advance!

**Reply by ptitSeb** [reply]
The app is loading gtk3?
Also, this syscall 328, I need to look at what it is.

Thoses are things that needs to be added to box86.

**Reply by Exclution** [reply]
@ptitSeb thank you for your reply. unfortunately i don't know if this app is loading gtk3 :-(
could it help if i provide more informations? can i do something to add those to box86?

**Reply by Exclution** [reply]
@ptitSeb should i open a case in "Pull request"?

**Reply by ptitSeb** [reply]
in "Pull request"? That is only if you do have a pull request to submit.
To see if the app use gtk3, use `BOX86_LOG=1` and see what gtk libs it loads...
To ask for the addition of the syscall or the symbol, open an issue ticket (but for the symbol, I need to know if it's gtk3 (wich is probably the case, as the symbol is already in gtk2, but it's better to confirm that, because error can come from elsewhere)

**Reply by Exclution** [reply]
@ptitSeb 
okay, here is the output but using BOX86_LOG=2:
Trying to add "libgtk-3.so.0" to maplib
Trying to load "libgtk-3.so.0"
Simplified name is "libgtk-3.so.0"
Using native(wrapped) libgtk-3.so.0
Trying to add "libgdk-3.so.0" to maplib
Trying to load "libgdk-3.so.0"
Simplified name is "libgdk-3.so.0"
Using native(wrapped) libgdk-3.so.0

see attached file for more information. 
[log.txt](https://github.com/ptitSeb/box86/files/6682164/log.txt)

i'll open an issue ticket. Thanks!

--> Issue ticket opened: https://github.com/ptitSeb/box86/issues/398

**Comment by chills340** [comment]
I've done it before by using the debian packages site to download the i386 libs
`https://packages.debian.org/sid/i386/libnss3/download`

and then:
`dpkg-deb -xv libnss3_3.67-1_i386.deb `
to extract it. you can then move the lib into the apps folder and it should detect it. 

just realized that's sid might need another version of that lib but the concept still applies

**Reply by Exclution** [reply]
Do you mean into the "squashfs-root" folder? this didnt worked for me... it stills shows libnss3.so is missing...

**Reply by chills340** [reply]
yes I could be wrong about where it goes, its been awhile since i've had to add extra  i386 libs for the stuff i run

**Reply by Exclution** [reply]
@chills340 after putting the lib files directly in the squashfs-root folder the libs were accessible, thanks for your suggestions 👍
